### ISSUE TITLE
Import sumo libraries in another cmake projects

### ISSUE THREAD

**Issue by alexampa** [eclipse/sumo#4485]
## Import sumo in a cmake project
Hello! I'm trying to import sumo, in particular libsumo, into a cmake project using command find_package(libsumostatic) in the corresponding CMakeLists file. However, when the project is being built the output is as follows:

```
...
Could not find a package configuration file provided by "libsumostatic"
  with any of the following names:

    libsumostaticConfig.cmake
    libsumostatic-config.cmake

  Add the installation prefix of "libsumostatic" to CMAKE_PREFIX_PATH or set
  "libsumostatic_DIR" to a directory containing one of the above files.  If
  "libsumostatic" provides a separate development package or SDK, be sure it
  has been installed.
...
```
Is there a way I could achieve importing the libsumo interface into my project? If not, could this be added as a feature? Thanks in advance!

### Specifications

  - Ubuntu version: 16.04
  - SUMO version: 0.32.0
  - SUMO branch: master

**Comment by namdre** [comment]
Please try with the latest development version instead of 0.32.0

**Comment by alexampa** [comment]
I'm sorry, I made a mistake about the version. I actually cloned the latest one (in particular 2853b1d), from the github repository and I followed the build instructions in the README file and in SUMO wiki for building and installing the SUMO binaries on Linux using cmake. The commands I used in detail are the following:
```
git clone --recursive https://github.com/eclipse/sumo
export SUMO_HOME="/path/to/sumo"
cd sumo && python tools/build/version.py
make -f Makefile.cvs
./configure && make install
mkdir build/cmake-build && cd build/cmake-build
cmake ../..
make
make install
```

**Comment by behrisch** [comment]
You are doing both build options (cmake and autools) at once, which is not necessary. You should have a working sumo already after the first make install. Or if you'd rather do the cmake build, the following should suffice:

    git clone --recursive https://github.com/eclipse/sumo
    export SUMO_HOME="/path/to/sumo"
    mkdir build/cmake-build && cd build/cmake-build
    cmake ../..
    make
    make install
Also usually you need to have root privileges to do make install (so you have to do `sudo make install`) but your error message does not look like this is the problem. Can you upload / post the complete build log somewhere?

**Comment by alexampa** [comment]
@behrisch 
Thanks for the reply. I do have root privileges and I don't have any problem building the sumo project, everything is built fine.

I just cannot import any sumo library in another cmake project, probably because the libraries are not exported during the build process of sumo. When I type find_package(libsumostatic) in a CMakeLists.txt in **another** project and try to build that project I get the following error:
```
Could not find the required component 'libsumostatic'. The following CMake error indicates that you either need to install the package with the same name or change your environment so that it can be found.
CMake Error at ...[...]... (find_package):
  Could not find a package configuration file provided by "libsumostatic"
  with any of the following names:

    libsumostaticConfig.cmake
    libsumostatic-config.cmake

  Add the installation prefix of "libsumostatic" to CMAKE_PREFIX_PATH or set
  "libsumostatic_DIR" to a directory containing one of the above files.  If
  "libsumostatic" provides a separate development package or SDK, be sure it
  has been installed.
```

I made a search in my file system and, indeed, there isn't any libsumostaticConfig.cmake nor libsumostatic-config.cmake file (which probably should have been generated after building the sumo project so as to import the libsumo library successfully).

**Comment by behrisch** [comment]
Sorry my fault, I misread your initial description. There is currently no cmake module for finding libsumo, but we can keep this open as an enhancement request. For now you will need to add the include path and the library manually to your build.

**Comment by rocktale** [comment]
Hi, I really like the idea of `libsumo` as far as I understand it (the documentation is still a bit scarce). It would make the integration into other C++ based simulation tools (omnet++, ns-3) much simpler and more efficient. However, as far as I see the static library and required headers are not exported (installed), yet. 

Pointing such tools to the sumo build folder (for the static lib) and the source folder (for the includes) is possible but hard to maintain (and brittle if paths change during versions). A shared library (libsumo) exported/installed along with the required headers and CMake modules for third party C++ software would greatly simplify this. Is there any plan for progress in that direction?

**Comment by behrisch** [comment]
Yes, we will probably do that once we have a project needing it but there is no specific date to that. Patches are always welcome ;-)

**Comment by aaron-bray** [comment]
Hi all,

I am working on this, and have a few questions

It looks like you are _only_ wanting users to use libsumocpp or libsumoc as a _shared_ target (dll/so)?
(This assumtion comes from SUMO only installing a shared libsumocpp, and everything else being static)

Do you want to also export/install all of the static libraries or not?
If I do this, _all_ static libs will also be installed during the build

I currently am able to add SUMO to my C++ exe with something like the following CMake
```add_executable(myApp ${files} SUMO::libsumocpp)```

Exporting/installing static lib would need to enumerate all required libs
```add_executable(myApp ${files} SUMO::libutil_common SUMO::libutil_vehicle ...)```

I am fine with just libsumocpp, but up to you
I don't have any issue with either way, up to how you want users to interact wih it
Even if the static libs are exported, you can still just link to the single shared lib (easiest usage)

Thoughts?

**Comment by behrisch** [comment]
I don't think anyone benefits from our static libs. They have weird interdependencies and are mostly a result of structuring the source code in directories but hardly individually usable. So let's focus on the dynamic libs.

**Comment by behrisch** [comment]
@aaron-bray BTW: Thanks for picking this one up!

**Comment by HuaweiAlgolux** [comment]
> add_executable

Hi, can you share your CMakeList.txt? I'm trying to write a c++ program that need to use sumo as a librarry. I'm struggling to make CMakeList.txt correct?

**Comment by aaron-bray** [comment]
Check out issue #8356

**Comment by behrisch** [comment]
I tried to do a small patch now which keeps the original filesystem layout (include/libsumo etc.) and it should now be possible to do 
find_package(SUMO REQUIRED)
target_link_libraries(MyTarget SUMO::libtracicpp)

Please give feedback

**Comment by behrisch** [comment]
Tests are still missing see #8356
